pidtree_bcc/probes/__init__.py

```diff
@@ -33,29 +39,33 @@ class variable defining a list of config fields.
                                   variable containing default templating variables.
         """
         self.output_queue = output_queue
-        self.plugins = load_plugins(probe_config.get('plugins', {}))
+        self.validate_config(probe_config)
         module_src = inspect.getsourcefile(type(self))
         self.probe_name = os.path.basename(module_src).split('.')[0]
+        self.plugins = load_plugins(probe_config.get('plugins', {}), self.probe_name)
         if not hasattr(self, 'BPF_TEXT'):
             with open(re.sub(r'\.py$', '.j2', module_src)) as f:
                 self.BPF_TEXT = f.read()
+        template_config = (
```

Comment:
This separates out the actual config injected into the bpf text from the probe meta-configuration then? Nice.